From: Raudi <mpell@web.de>
Date: Mon, 6 Jan 2014 19:44:08 -0500
Subject: hwdb: change key mappings for Samsung 90X3A

The Key codes didn't match with the described key. Also the key
identifier strings were missing. I hope I chose appropriate ones.

https://bugs.freedesktop.org/show_bug.cgi?id=70296
(cherry picked from commit dfa2ea215df5f1e78084de862a54c0f1f321a13a)
---
 hwdb/60-keyboard.hwdb | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hwdb/60-keyboard.hwdb b/hwdb/60-keyboard.hwdb
index ab9e569..644cb94 100644
--- a/hwdb/60-keyboard.hwdb
+++ b/hwdb/60-keyboard.hwdb
@@ -886,7 +886,6 @@ keyboard:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*940X3G*:pvr*
  KEYBOARD_KEY_b3=!prog3                                 # Fn+F11 fan/cooling mode changer
 
 # Series 9
-keyboard:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*90X3A*:pvr*
 keyboard:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*900X[34][AB]*:pvr*
  KEYBOARD_KEY_ce=!                                      # Fn+F8 keyboard backlight up
  KEYBOARD_KEY_8d=!                                      # Fn+F7 keyboard backlight down
@@ -894,6 +893,13 @@ keyboard:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*900X[34][AB]*:pvr
  KEYBOARD_KEY_97=!                                      # Fn+F12 Wi-Fi toggle
  KEYBOARD_KEY_d5=!                                      # Fn+F6 battery life extender
 
+keyboard:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pn*90X3A*:pvr*
+ KEYBOARD_KEY_ce=!prog1                                 # Fn+F1 launch settings
+ KEYBOARD_KEY_8d=!prog3                                 # Fn+F6 performance mode
+ KEYBOARD_KEY_97=!kbdillumdown                          # Fn+F7 keyboard backlight down
+ KEYBOARD_KEY_96=!kbdillumup                            # Fn+F8 keyboard backlight up
+ KEYBOARD_KEY_d5=!wlan                                  # Fn+F12 Wi-Fi toggle
+
 # SQ1US
 keyboard:dmi:bvn*:bvr*:bd*:svn[sS][aA][mM][sS][uU][nN][gG]*:pnSQ1US:pvr*
  KEYBOARD_KEY_d4=menu
